provider/src/main/kotlin/org/gradle/kotlin/dsl/ExtraPropertiesExtensions.kt

```diff
@@ -39,7 +40,12 @@ operator fun <T> ExtraPropertiesExtension.setValue(receiver: Any?, property: KPr
 
 
 operator fun <T> ExtraPropertiesExtension.getValue(receiver: Any?, property: KProperty<*>): T =
-    uncheckedCast(get(property.name))
+    property.run {
+        val isFound = has(name)
```

Comment:
:o: Given how [`DefaultExtraPropertiesExtension`](https://github.com/gradle/gradle/blob/f3410bf22080a6ea5f9f6c5a16aa78c776e9d2e0/subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultExtraPropertiesExtension.java#L39) works, this implementation will cause every Kotlin delegate property access to an existing property cost `2 * costOf(Map<*, *>::containsKey) + costOf(Map<*, *>::get) + costOf(KProperty<*>::returnType.isMarkedNullable)` which I am slightly uncomfortable with.

A `ExtraPropertiesExtension.provideDelegate` operator could alleviate some of these costs.

Reply:
Right. A `provideDelegate` operator would allow to alleviate `costOf(KProperty<*>::returnType.isMarkedNullable)`. But in order to short circuit `containsKey` we would need a `ExtraPropertiesExtension.find()` method.
I'll add a delegate provider to remove `returnType.isMarkedNullable` access on each delegated property access, it'll already be better.